analyses/survival-analysis/02-summary-statics-TP53.Rmd

```diff
@@ -293,6 +306,71 @@ ggsave(file.path(plots_dir, "boxplot_tp53_score_race.pdf"))
 
 ```
 
+### The distribution of TP53 score among predicted ancestry
+
+```{r}
+group_summary <- HGG_survival %>%
+  group_by(predicted_ancestry) %>%
+  summarize(count = n())
+
+# Filter the groups with at least 3 samples
+HGG_filtered <- HGG_survival %>%
+  filter(predicted_ancestry %in% group_summary$predicted_ancestry[group_summary$count >= 3])
+
+p1 <- HGG_filtered %>% 
+  mutate(predicted_ancestry = fct_reorder(predicted_ancestry, -tp53_score, .fun='median' )) %>%
+  ggplot(aes(x = predicted_ancestry, y = tp53_score)) + 
+  geom_boxplot() +
+  geom_jitter(width = 0.3, alpha = 0.5) + 
+  xlab("") +
+  stat_compare_means(method = "wilcox", comparisons = list(c("EAS", "AFR"), c("EAS", "EUR"), c("EAS", "AMR"), c("EAS", "Not reported"))) +
+  stat_compare_means(method = "kruskal", label.y = 1.6) + 
+  theme_Publication() + 
+  ylab("TP53 score") + 
+  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
+
+p1
+```
+
+### save the plot
+
+```{r}
+ggsave(file.path(plots_dir, "boxplot_tp53_score_ancestry.pdf"))
+
+```
+
+## alluvial plot for race and predicted-ancestry
+
+```{r}
+HGG_race <- as.data.frame(table(HGG_survival$race, HGG_survival$predicted_ancestry)) %>% 
+  dplyr::rename(race = Var1, 
+                predicted_ancestry = Var2) %>%
+  to_lodes_form(axes = 1:2) %>% 
+  dplyr::rename(Group = stratum) %>% 
+  filter(Group != "Not reported") %>% 
+  mutate(Group = factor(Group, levels = c("Asian", "Black or African American", "Native Hawaiian or Other Pacific Islander", "Other", "Reported Unknown", "White", "EAS", "SAS", "AFR", "EUR", "AMR")))
+
+p1 <- ggplot(HGG_race, aes(y = Freq, stratum = Group, alluvium = alluvium, x= x, fill = Group)) + 
+  geom_alluvium() + 
+  geom_stratum() +
+  scale_fill_manual(values = c("Asian" = "darkorange", "SAS" = "goldenrod1", "EAS" = "tan", 
+                                "White" = "firebrick4", "EUR" = "lightpink", 
+                                "Black or African American" = "royalblue4", "AFR" = "steelblue1", 
+                                "Native Hawaiian or Other Pacific Islander" = "olivedrab", "AMR" = "orchid4", 
+                                "Reported Unknown" = "black", "Other" = "ivory4")) + 
+  xlab("") + 
+  theme_Publication()
```

Comment:
Is it possible to have two separate legends here, one for reported race and another for predicted ancestry? And then, as Jo Lynne suggested, make the colors similar between "matched" race and ancestry groups.

Reply:
Thank you for the suggestion! It's done